src/group_theory/coset.lean

```diff
@@ -3,19 +3,45 @@ Copyright (c) 2018 Mitchell Rowett. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Mitchell Rowett, Scott Morrison
 -/
+
 import group_theory.subgroup
 
+/-!
+# Cosets
+
+This file develops the basic theory of left and right cosets.
+
+## Main definitions
+
+* `left_coset a s`: the left coset `a * s` for an element `a : α` and a subset `s ⊆ α`.
+* `right_coset s a`: the right coset `s * a` for an element `a : α` and a subset `s ⊆ α`.
+* `quotient s`: the quotient type representing the left cosets with respect to a subgroup `s`.
+* `mk`: the canonical map from `α` to `α/s` for a subgroup `s` of `α`.
```

Comment:
I'm unsure about whether these should be fully qualified names, that is `quotient_group.quotient` and `quotient_group.mk` or not. One benefit I can think of for using fully qualified names here is that doc-gen would generate correct links (which IMO means this should be changed).

In that case, it should probably be mentioned somewhere that there are `add_` versions of everything.